augur/io/metadata.py

```diff
@@ -424,3 +426,47 @@ def write_records_to_tsv(records, output_file):
 
         for record in records:
             tsv_writer.writerow(record)
+
+
+class Metadata(TabularFile):
+    def __init__(self, file:str,
+            header:Optional[bool]=True,
+            names:Optional[List[str]]=None,
+            potential_id_columns:Optional[List[str]]=None):
+        super().__init__(file, header, names)
+
+        self.potential_id_columns = potential_id_columns or ("strain", "name")
+        """Metadata columns that may contain identifier information."""
+
+        self.id_column = _get_first_valid_column(self.file, self.potential_id_columns)
+        """Metadata ID column name."""
+
+
+def _get_first_valid_column(metadata_file:str, potential_column_names:List[str]):
```

Comment:
What if this function and `_get_column_names` were implemented as `classmethod`s instead of as "private" top-level functions in the module? Since we're defining a `Metadata` class already, it makes sense to me to bundle the related class methods with the class where they can be inherited or more easily discovered. They could still be implemented as "private" methods with the underscore notation, but reorganizing them as `classmethod`s would also communicate the same sense of internal functionality.

Reply:
Thanks for flagging. This (along with many other things) are artifacts of extracting code from the other PR.

I agree more of these lingering functions should be class methods. Done: 070b8b4fbe45925aada0a3585108563b4056c84a, 87568955fe2753562bd94789f01dddb7c6413b3a. This also enabled bf9f551ec5dc56d2787ba34c77d2bd71ce045c8f for a cleaner implementation.